python_dwd/file_path_handling/file_list_creation.py

```diff
@@ -1,75 +1,46 @@
 """ file list creation for requested files """
-from pathlib import Path
-from typing import List, Union
+from typing import Union
 import pandas as pd
 
-from python_dwd.additionals.functions import check_parameters
-from python_dwd.additionals.helpers import create_fileindex
-from python_dwd.constants.access_credentials import DWD_FOLDER_MAIN, DWD_FOLDER_METADATA
-from python_dwd.constants.metadata import FILELIST_NAME, DATA_FORMAT
 from python_dwd.enumerations.column_names_enumeration import DWDMetaColumns
 from python_dwd.enumerations.parameter_enumeration import Parameter
 from python_dwd.enumerations.period_type_enumeration import PeriodType
 from python_dwd.enumerations.time_resolution_enumeration import TimeResolution
+from python_dwd.file_path_handling.file_index_creation import create_file_index_for_dwd_server, \
+    reset_file_index_cache
 
 
-def create_file_list_for_dwd_server(station_ids: List[int],
+def create_file_list_for_dwd_server(station_id: Union[str, int],
```

Comment:
please avoid breaking changes here. why do you downgrad this function?

Reply:
The function was never really intended to create a file list for more then one station id. I had changed this to make it compatible for creating the **HAS_FILE** column of the resulting metadata DataFrame. However now with the new file index I find it to be more clean to instead call the file index and then work with that complete set instead.

Reply:
But this function worked fine for my use case parsing multiple system id's. 

Can you explain the idea of the file index? Does this list all files on the server? (= all stations)

Reply:
So I see no usecase for this function as a stand alone returning files for multiple station ids. For this case we created the pipeline which is able to work on several station ids...

The file index represents the available files on the server side instance for a given set of parameters e.g. "kl", "daily", "historical". From that filenames can be gathered for an requested station id.

Reply:
I reverted this to the previous state.